tobias/utils/motifs.py

```diff
@@ -851,6 +880,106 @@ def create_logo(self, ax = None, motif_len=None):
 
 		return logo
 
+	def set_counts(self, counts):
+		""" 
+		Set the count matrix. Also updates number of sites (n) and motif length (length).
+
+		Parameters:
+			counts (list): List of length 4. Contains lists of base counts for each position.
+
+		Returns:
+			self (OneMotif): OneMotif object with modified counts.
+  		"""
+    	# check counts input
+		if len(counts) != 4:
+			raise ValueError("Input counts must be of length 4. Received length {0}".format(len(counts)))
+		lengths = [len(base) for base in counts]
+		if len(set(lengths)) != 1:
+			raise ValueError("All lists in counts must be of same length.")
+
+		# add counts
+		self.counts = counts
+		# update motif length
+		self.length = lengths[0]
+		# update number of sites
+		self.n = np.sum([row[0] for row in counts])
+
+		return self
+
+	def as_string(self, output_format="pfm", header=True):
+		"""
+		Returns the OneMotif object as a string in the given output_format.
+  
+		Parameters:
+			output_format (string): Set the output format one of ["pfm", "jaspar", "meme"].
+
+			header (bool): Whether a header should be generated (meme only).
+   
+		Returns:
+			(string): String in the chosen file format.
+		"""
+		out_string = ""
+  
+		if output_format in ["pfm", "jaspar"]:
+			# create id line
+			out_string += f">{self.id}\t{self.name}\n"
+
+			# add counts
+			for index, base in enumerate(self.bases):
+				row = " ".join(map(str, self.counts[index]))
+
+				if output_format == "jaspar":
+					row = f"{base} [{row} ]"
+
+				out_string += row + "\n"
+			
+		elif output_format == "meme":
+			# create meme header
+			if header:
+				# TODO implement for newer version
+				meme_header = "MEME version 4\n\n"
+				meme_header += "ALPHABET= {0}\n\n".format("".join(self.bases))
+				meme_header += "strands: {0}\n\n".format(self.strand if self.strand != "." else "+ -")

```

Comment:
Hmm, I am not sure about this. Since motifs usually don't have a strand, self.strand is only set to use in `setup_moods_scanner` and `scan_sequence` to know which strand the hit came from. The MEME format (as explained [here](http://meme-suite.org/doc/meme-format.html)) states that the `strand:` indicates "if motifs were created from sites on both the given and the reverse complement strands of the sequences". Basically, "the strand from which it was created from" and "the strand which should be searched" are two different things. Also now, you take the first motif to decide this parameter for the whole list. I don't know how to resolve this at the moment, but let me know your thoughts!

Reply:
So it is the opposite between moods and meme? Hm, I think we should reverse the strand for moods when the scanner is created as the strand most likely will be coming from a meme file. Else it might be confusing what the strand is telling if we reverse it too often.
Using the strand from the first motif when writing multiple is a problem. The same goes also for the background. But I would say this comes close to a design issue with the meme format itself. As a solution for us, we could implement a `get_strand` function that is similar to `get_background` and create the header in MotifList.